framework/widgets/ActiveField.php

```diff
@@ -738,7 +738,9 @@ protected function getClientOptions()
                 $js = $validator->clientValidateAttribute($this->model, $attribute, $this->form->getView());
                 if ($validator->enableClientValidation && $js != '') {
                     if ($validator->whenClient !== null) {
-                        $js = "if (({$validator->whenClient})(attribute, value)) { $js }";
+                        $exp = is_callable($validator->whenClient) ?
```

Comment:
The method was already over-complicated with nested `if` conditions, adding one more does not make the method better :(